code/modules/food_and_drinks/kitchen_machinery/microwave.dm

```diff
@@ -287,8 +287,8 @@
 
 /obj/machinery/microwave/proc/loop(type, time, wait = max(12 - 2 * efficiency, 2)) // standard wait is 10
 	if(machine_stat & (NOPOWER|BROKEN))
-		if(type == MICROWAVE_PRE)
-			pre_fail()
+		pre_fail()
+		eject()
```

Comment:
I think, better place this in power_change proc.
https://github.com/tgstation/tgstation/blob/586bc81b4a0326d6b8b35fa30a086496621d01f4/code/modules/power/power.dm#L113-L134
Or use signals.